lib/action_view/component/base.rb

```diff
@@ -55,7 +53,13 @@ def render_in(view_context, *args, &block)
         @content = view_context.capture(&block) if block_given?
         validate!
 
-        send(self.class.call_method_name(@variant))
+        call_method_name = if self.respond_to?(self.class.call_method_name(@variant))
```

Comment:
This decision tree ends up being pretty similar to `.call_method_name`. Might it be possible to combine the two?